packages/flutter/lib/src/services/asset_vendor.dart

```diff
@@ -0,0 +1,60 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+import 'dart:async';
+
+import 'package:mojo/core.dart' as core;
+
+import 'asset_bundle.dart';
+import 'asset_resolver.dart';
+
+/// Configuration parameters for a particular AssetResolver.
+///
+/// Since AssetVendor uses configuration objects as cache keys, subclasses of
+/// AssetResolverConfig should override operator== and hashCode so that
+/// equivalent configurations appear to be equal. Config objects should also be
+/// cheap to construct.
+abstract class AssetResolverConfig {
+  const AssetResolverConfig();
+  AssetResolver createResolver();
+  operator==(AssetResolverConfig config);
+  int get hashCode;
+}
+
+/// Asset bundle capable of vending assets via the resolution logic of an
+/// [AssetResolver].
+///
+/// Each AssetVendor is specific to a particular underlying asset bundle and
+/// asset resolver configuration. To retrieve an appropriate AssetVendor, call
+/// the static `vendorFor` method.
+class AssetVendor extends CachingAssetBundle {
+
+  static Map<AssetResolverConfig, AssetVendor> _cache =
+    <AssetResolverConfig, AssetVendor>{};
+
+  /// Retrieve an asset vendor for the given asset bundle and asset resolver
+  /// configuration. If no such vendor exists yet, one will be created via
+  /// the configuration's `createResolver` method.
+  static AssetVendor vendorFor(AssetBundle bundle, AssetResolverConfig config) {
+    return _cache.putIfAbsent(config, () {
+      return new AssetVendor(bundle: bundle, resolver: config.createResolver());
+    });
+  }
+
+  AssetVendor({ this.bundle, this.resolver });
```

Comment:
Actually never mind. The statics you have here make sense to come before the constructor since that's the order they'd be called in.